src/mainboard/emulation/qemu-q35/src/main.rs

```diff
@@ -26,26 +33,35 @@ pub extern "C" fn _start(fdt_address: usize) -> ! {
     let kernel_segs = &[
         payload::Segment {
             typ: payload::stype::PAYLOAD_SEGMENT_ENTRY,
-            base: 0x80000000,
-            data: &mut SectionReader::new(&Memory {}, 0x20000000 + 0x400000, 6 * 1024 * 1024),
+            base: 0x1000020,
+            // wtf. data should not be mutable?
```

Comment:
I see the problem. It looks like we made the unfortunate design decision that the entry point is at the beginning of a segment (which worked so far). That should be changed.